### PR TITLE
lab: generate and install zsh completion

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

zsh completion! :)

Env variables are set to dummy values
so `lab` doesn't get stuck trying to helpfully
interact with user for initial configuration.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/version-management/git-and-tools/lab/default.nix [hunk 1]
```diff
@@ -17,6 +17,12 @@ buildGoPackage rec {
 
   goDeps = ./deps.nix;
 
+  postInstall = ''
+    mkdir -p $out/share/zsh/site-functions
+    LAB_CORE_HOST=a LAB_CORE_USER=b LAB_CORE_TOKEN=c \
+    $out/bin/lab completion zsh > $out/share/zsh/site-functions/_lab
```
Comment (worldofpeace) [inline pkgs/applications/version-management/git-and-tools/lab/default.nix]:
```suggestion
    $bin/bin/lab completion zsh > $bin/share/zsh/site-functions/_lab
```

**Comment by dtzWill** [comment]
Oh! Ha, well this is because I forgot to upstream porting `lab` over to
the new `buildGoModule` goodness.
Which apparently has different behavior re:outputs?
Doesn't matter here, but that's why $out worked for me :3.

Will address in a few.

**Comment by worldofpeace** [comment]
> Oh! Ha, well this is because I forgot to upstream porting `lab` over to the new `buildGoModule` goodness. Which apparently has different behavior re:outputs?

Hmm, I wasn't aware of `buildGoModule` (so many happenings in nixpkgs) but I'm presently surprised :smile: 

@kalbasit Any reason why the outputs for `buildGoModule` are firstly `out` wherein `buildGoPackage` it was `bin`?

**Comment by dtzWill** [comment]
> @kalbasit Any reason why the outputs for `buildGoModule` are firstly `out` wherein `buildGoPackage` it was `bin`?

It seems it's more that `buildGoModule` has a single output by default ("out"), where `buildGoPackage` would have two -- which I'm sure there's a reason for but I don't know it :).  May be good to sort this out or make them consistent, dunno.

Anyway, merging this as it should be ready now!

**Comment by dtzWill** [comment]
Just kidding, hashes are wrong? They might depend on something I have changed, will check sorry.

**Comment by worldofpeace** [comment]
Weird they're still wrong. :confused:

**Comment by kalbasit** [comment]
> @kalbasit Any reason why the outputs for `buildGoModule` are firstly `out` wherein `buildGoPackage` it was `bin`?

The initial version of buildGoModule had both the $bin and the $out outputs. However, @Mic92 pointed out that all of the buildGoPackage derivations had a useful output ($bin) and a rather empty and useless output $out and he was unable to consolidate both due to all packages (and possibly outside nixpkgs) usage of buildGoPackage. Since buildGoModule was a new infrastructure, we wanted to consolidate the outputs.

> Hmm, I wasn't aware of `buildGoModule` (so many happenings in nixpkgs) but I'm presently surprised

https://kalbas.it/2019/03/17/announcing-the-new-golang-infrastructure-buildgomodule/ :)

**Comment by worldofpeace** [comment]
> The initial version of buildGoModule had both the $bin and the $out outputs. However, @Mic92 pointed out that all of the buildGoPackage derivations had a useful output ($bin) and a rather empty and useless output $out and he was unable to consolidate both due to all packages (and possibly outside nixpkgs) usage of buildGoPackage. Since buildGoModule was a new infrastructure, we wanted to consolidate the outputs.

Right, all outputs should provide some sort of use. Think you made the correct decision there.

> https://kalbas.it/2019/03/17/announcing-the-new-golang-infrastructure-buildgomodule/ :)

Nice writeup :+1: Thank you for your contributions to the Golang infrastructure.

File: pkgs/applications/version-management/git-and-tools/lab/default.nix [hunk 2]
```diff
@@ -1,27 +1,31 @@
...
     owner = "zaquestion";
     repo = "lab";
     rev = "v${version}";
     sha256 = "1210cf6ss4ivm2jxq3k3c34vpra02pl91fpmvqbvw5sm53j7xfaf";
   };
 
-  goDeps = ./deps.nix;
+  subPackages = [ "." ];
+
+  modSha256 = "0bw47dd1b46ywsian2b957a4ipm77ncidipzri9ra39paqlv7abb";
```
Comment (kalbasit) [inline pkgs/applications/version-management/git-and-tools/lab/default.nix]:
```suggestion
  modSha256 = "1g72fnkibmdd9i20wgp5fjf556xp2xwan06dmn3f5rkmnicy3ccl";
```

as reported by ofborg.

**Review by kalbasit — DISMISSED** [review]

**Pull request merged**